frontends/web/.eslintrc.js

```diff
@@ -0,0 +1,214 @@
+  "plugins": ["@typescript-eslint"],
+  "extends": [
+    "eslint:recommended",
+    "plugin:@typescript-eslint/eslint-recommended",
+    "plugin:@typescript-eslint/recommended",
+    "plugin:@typescript-eslint/recommended-requiring-type-checking",
+    "plugin:react/recommended"
+  ],
+  "parserOptions": {
+    "ecmaVersion": "8",
+    "sourceType": "module",
+    "ecmaFeatures": {
+      "jsx": true
+    },
+    "project": require.resolve("./tsconfig.json"),
+  },
+  "settings": {
+    "react": {
+        "pragma": "h",
+        "version": "16"
+    },
+  },
+  "ignorePatterns": [
+    "*.css.d.ts"
+  ],
+  "rules": {
+    "react/display-name": [2, { ignoreTranspilerName: false }],
+    // "react/jsx-equals-spacing": 2,
+    "react/jsx-equals-spacing": "off",
+    "react/jsx-no-comment-textnodes": 2,
+    "react/jsx-no-duplicate-props": 2,
+    // "react/jsx-no-target-blank": 2,
+    "react/jsx-no-target-blank": "off",
+    "react/jsx-no-undef": 2,
+    "react/jsx-pascal-case": 2,
+    "react/jsx-uses-react": 2,
+    "react/jsx-uses-vars": 2,
+    // "react/jsx-wrap-multilines": 2,
+    "react/jsx-wrap-multilines": "off",
+    "react/no-danger": 2,
+    // "react/no-did-mount-set-state": 2,
+    "react/no-did-mount-set-state": "off",
+    // "react/no-did-update-set-state": 2,
+    "react/no-did-update-set-state": "off",
+    "react/no-find-dom-node": 2,
+    "react/no-is-mounted": 2,
+    "react/no-string-refs": 0,
+    "react/prefer-es6-class": 2,
+    // "react/prefer-stateless-function": 1,
+    "react/prefer-stateless-function": "off",
+    "react/require-render-return": 2,
+
+    "no-eval": 2,
+    "no-implied-eval": 2,
+    "no-new-func": 2,
+
+    "arrow-parens": [0, "always"],
+    "arrow-spacing": 2,
+    camelcase: [1, { properties: "never" }],
+    "comma-style": [2, "last"],
+    "constructor-super": 2,
+    curly: [0, "multi-line"],
+    "dot-notation": [2, { allowPattern: "^[a-z]+(_[a-z]+)+$" }],
+    eqeqeq: 2,
+    "func-style": [2, "declaration", { allowArrowFunctions: true }],
+    "guard-for-in": 2,
+    "handle-callback-err": 0,
+    "jsx-quotes": [2, "prefer-double"],
+    // "key-spacing": 2,
+    "key-spacing": "off",
+    "keyword-spacing": 2,
+    // "lines-around-comment": 2,
+    "lines-around-comment": "off",
+    "new-cap": 1,
+    "new-parens": 2,
+    "no-array-constructor": 2,
+    "no-caller": 2,
+    "no-cond-assign": 2,
+    // "no-console": [1, { allow: ["warn", "error"] }],
+    "no-console": "off",
+    "no-const-assign": 2,
+    "no-delete-var": 2,
+    "no-dupe-class-members": 2,
+    "no-dupe-keys": 2,
+    // "no-duplicate-imports": 2,
+    "no-duplicate-imports": "off",
+    // "no-else-return": 2,
+    "no-else-return": "off",
+    "no-empty-pattern": 0,
+    "no-empty": 0,
+    "no-extra-parens": 0, // we use extra parens in JSX ternaries, no way to disable that here.
+    "no-iterator": 2,
+    // "no-lonely-if": 2,
+    "no-lonely-if": "off",
+    "no-mixed-spaces-and-tabs": [2, "smart-tabs"],
```

Comment:
i'd comment this for now, but do without 'smart-tabs' later. seems like smart-tabs allows pretty much everything.

Reply:
https://github.com/digitalbitbox/bitbox-wallet-app/pull/1013